ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java

```diff
@@ -128,17 +132,26 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
             {
                 List<HarvestEvent> events = this.harvestService.harvest(this.harvestSize);
 
-                if(events.size() > 0 && dashboardRestService.publish(events))
+                if(events.size() > 0)
                 {
-                    harvestService.updateAsHarvested(events);
+                    if(dashboardRestService.publish(events))
+                    {
+                        harvestService.updateAsHarvested(events);
+                    }
+                    else if(this.monitor!=null)
+                    {
+                        this.monitor.invoke(HarvestJobState.ERROR);
+                    }
                 }
             }
 
+            if(this.monitor!=null)this.monitor.invoke(HarvestJobState.HEALTHY);
```

Comment:
I think the condition should be moved under line 139. otherwise you run the risk of:
this.monitor.invoke(HarvestJobState.ERROR);
this.monitor.invoke(HarvestJobState.HEALTHY);

Reply:
this is sorted